src/Product/changeDetails.ts

```diff
@@ -1,51 +1,44 @@
 import assert from 'assert';
+import { omit } from 'lodash';
 import { HttpModule } from '@whppt/api-express';
 import { Product } from './Models/Product';
 
-const changeDetails: HttpModule<Product, { status: number }> = {
-  authorise() {
-    return Promise.resolve();
+const changeDetails: HttpModule<Product, void> = {
+  authorise({ $identity }, { user }) {
+    return $identity.isUser(user);
   },
-  exec({ $salesforce, $mongo: { $dbPub, $saveToPubWithEvents, $startTransaction }, createEvent }, productData) {
+  exec({ $database, createEvent }, productData) {
     assert(productData.domainId, 'Product requires a Domain Id.');
     assert(productData._id, 'Product requires a Product Id.');
     assert(productData.name, 'Product requires a Name.');
     assert(productData.productCode, 'Product requires a Product Code.');
 
-    return $dbPub
-      .collection('products')
-      .findOne<Product>({
-        _id: productData._id,
-      })
-      .then(product => {
+    return $database.then(({ document, startTransaction }) => {
+      return document.fetch<Product>('products', productData._id).then(product => {
         assert(product, 'Product does not exsist');
         const event = createEvent('ProductDetailsChanged', productData);
-        Object.assign(product, productData);
-
-        return $startTransaction(session => {
-          return $saveToPubWithEvents('products', product, [event], { session }).then(() => {
-            return $salesforce.$Oauth().then((token: string) => {
-              return $salesforce.$upsert(token, product._id, salesForceItem(product));
-            });
-          });
-        }).then(() => ({ status: 200 }));
+        Object.assign(product, omit(productData, ['unleashed']));
```

Comment:
Should this be config?
If its just to clean up unleashed data we can probably remove it.
I do like omitting the config here though